### PR TITLE
converted sparse/umfpack to testsets

### PR DESCRIPTION
**Pull request by lbollar**
Referencing [#18801](https://github.com/JuliaLang/julia/issues/18801)

For the test set labeled very generically umfpack tests, I believe these were the tests added by @andreasnoack [here](https://github.com/JuliaLang/julia/commit/5becd8cddd599f587a1377f019aa598c16ba8c85) that began test file. I wasn't sure what to name these as they are unfamiliar to me and the first big chunk seems pretty generic, so if anyone has any suggestions for naming I would gladly change.

### CODE REVIEW

File: test/sparse/umfpack.jl [hunk 1]
```diff
@@ -1,156 +1,169 @@
...
+            L, U, p, q, Rs = F[:(:)]
+            @test (Diagonal(Rs) * A)[p,q] ≈ L * U
+        end
     end
 end
 
-#4523 - complex sparse \
-x = speye(2) + im * speye(2)
-@test (x*(lufact(x) \ ones(2))) ≈ ones(2)
+@testset "issue #4523 complex sparse" begin
```
Comment (tkelman) [inline test/sparse/umfpack.jl]: this should still say \

File: test/sparse/umfpack.jl [hunk 2]
```diff
@@ -1,156 +1,169 @@
...
-# B::StridedMatrix{T}) works as expected.
-let N = 10, p = 0.5
-    A = N*speye(N) + sprand(N, N, p)
-    X = zeros(Complex{Float64}, N, N)
-    B = complex.(rand(N, N), rand(N, N))
-    luA, lufA = lufact(A), lufact(Array(A))
-    @test A_ldiv_B!(copy(X), luA, B) ≈ A_ldiv_B!(copy(X), lufA, B)
-    @test At_ldiv_B!(copy(X), luA, B) ≈ At_ldiv_B!(copy(X), lufA, B)
-    @test Ac_ldiv_B!(copy(X), luA, B) ≈ Ac_ldiv_B!(copy(X), lufA, B)
+@testset "# Test that A[c|t]_ldiv_B!{T<:Complex}(X::StridedMatrix{T}, lu::UmfpackLU{Float64}, # B::StridedMatrix{T}) works as expected." begin
```
Comment (tkelman) [inline test/sparse/umfpack.jl]: remove the # characters from the description string

Comment (tkelman) [inline test/sparse/umfpack.jl]: remove the # characters from the description string

Comment (KristofferC) [inline test/sparse/umfpack.jl]: I did that

Comment (tkelman) [inline test/sparse/umfpack.jl]: there are 2 of them and you only got rid of one

Comment (KristofferC) [inline test/sparse/umfpack.jl]: where is the second one?

Comment (StefanKarpinski) [inline test/sparse/umfpack.jl]: I also do not see a second one – there are some testset strings with "issue #123" in them, but that's fine.

Comment (tkelman) [inline test/sparse/umfpack.jl]: `lu::UmfpackLU{Float64}, # B::StridedMatrix{T})`

Comment (fredrikekre) [inline test/sparse/umfpack.jl]: `# B::StridedMatrix{T}`: https://github.com/JuliaLang/julia/pull/20956/files#diff-7d74c26aa56c59a6099dc1408c2753d6R159

Comment (KristofferC) [inline test/sparse/umfpack.jl]: its gone now

**Comment by fredrikekre** [comment]
#21904

**Pull request closed without merging**